scripts/xtensa-build-zephyr.py

```diff
@@ -0,0 +1,399 @@
+	:rtype: [type] bool, pathlib.Path
+	"""
+	global west_top, SOF_TOP
+	paths = [ pathlib.Path(os.getcwd()), SOF_TOP, west_top]
+	for path in paths:
+		if path.is_dir():
+			result = execute_command(["west", "topdir"], stderr=subprocess.DEVNULL,
+				stdout=subprocess.DEVNULL, check=False, cwd=path)
+		if result.returncode == 0:
+			return path
+	return None
+
+def create_zephyr_directory():
+	global west_top
+	west_top.mkdir(mode=511, parents=False, exist_ok=False)
+	west_top = west_top.resolve(strict=True)
+
+def create_zephyr_sof_symlink():
+	global west_top, SOF_TOP
+	audio_modules_dir = pathlib.Path(west_top, "modules", "audio")
+	audio_modules_dir.mkdir(mode=511, parents=True, exist_ok=True)
+	sof_symlink = pathlib.Path(audio_modules_dir, "sof")
+	# Symlinks creation requires administrative privileges in Windows or special user rights
+	try:
+		if not sof_symlink.is_symlink():
+			sof_symlink.symlink_to(SOF_TOP, target_is_directory=True)
+	except:
+		print(f"Failed to create symbolic link: {sof_symlink} to {SOF_TOP}."
+			"\nIf you run script on Windows run it with administrative privileges or\n"
+			"grant user symlink creation rights -"
+			"see: https://docs.microsoft.com/en-us/windows/security/threat-protection/"
+			"security-policy-settings/create-symbolic-links")
+		raise
+
+def west_init_update():
+	"""[summary] Downloads zephyrproject inside sof/ and create a ../../.. symbolic
+	link back to sof/"""
+	global west_top, SOF_TOP
+	zephyr_dir = pathlib.Path(west_top, "zephyr")
+	execute_command(["git", "clone", args.url, str(zephyr_dir)], check=True, timeout=1200)
+	execute_command(["git", "fetch", "origin", args.z], check=True, timeout=300, cwd=zephyr_dir)
+	execute_command(["git", "checkout", "FETCH_HEAD"], check=True, cwd=zephyr_dir)
+	execute_command(["git", "-C", str(zephyr_dir), "--no-pager", "log", "--oneline", "--graph",
+		"--decorate", "--max-count=20"], check=True)
+	execute_command(["west", "init", "-l", str(zephyr_dir)], check=True)
+	create_zephyr_sof_symlink()
+	# Do NOT "west update sof"!!
+	execute_command(["west", "update", "zephyr", "hal_xtensa"], check=True, timeout=300, cwd=west_top)
+
+def build_platforms():
+	global west_top, SOF_TOP
+	# double check that west workspace is initialized
+	execute_command(["west", "topdir"], check=True, stderr=subprocess.DEVNULL,
+		stdout=subprocess.DEVNULL, cwd=west_top)
+	print(f"SOF_TOP={SOF_TOP}")
+	print(f"west_top={west_top}")
+	global STAGING_DIR
+	STAGING_DIR = pathlib.Path(west_top, "build-sof-staging", "sof")
+	# smex does not use 'install -D'
+	STAGING_DIR.mkdir(mode=511, parents=True, exist_ok=True)
+	for platform in args.platforms:
+		platform_dict = [x for x in platform_list if x["name"] == platform][0]
+		if "XTENSA_TOOLS_VERSION" in platform_dict:
+			XTENSA_TOOLS_ROOT = os.getenv("XTENSA_TOOLS_ROOT")
+			if not XTENSA_TOOLS_ROOT:
+				raise RuntimeError(f"Platform {platform} uses Xtensa toolchain."
```

Comment:
This makes it impossible to build with the Zephyr SDK.

So far we've been able to build every platform with either the Zephyr SDK or XCC, both work fine. The Zephyr SDK requires zero license and zero configuration, install it in the HOME directory and that's it, job done. Nothing to do in this script to support it (besides removing this RunTimeError)